resource-management/test/integration/framework/service.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2022 Authors of Global Resource Service.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	apiapp "global-resource-service/resource-management/cmds/service-api/app"
+	localMetrics "global-resource-service/resource-management/pkg/common-lib/metrics"
+	"global-resource-service/resource-management/pkg/store/redis"
+	simapp "global-resource-service/resource-management/test/resourceRegionMgrSimulator/app"
+	"global-resource-service/resource-management/test/resourceRegionMgrSimulator/data"
+)
+
+// ServiceMain to start gcs service and simulator service for testing.
+func ServiceMain(tests func() int) {
+
+	//flush redis store to ensure all testing started with clean store
+	redis.NewRedisClient("localhost", "7379", true)
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		masterIp := "localhost"
+		masterPort := "8080"
+		urls := "localhost:9119"
+		redisPort := "7379"
+		enableMetrics := false
+		grs_err := Start_ServiceAPI(masterIp, masterPort, urls, redisPort, enableMetrics)
+		if grs_err != nil {
+			klog.Infof("Starting resource management service failed with error: %v", grs_err)
```

Comment:
Do you need to fail the test if Start_ServiceAPI and/or Start_Simulator returns error?

Reply:
we can add this later if necessary.